File: 1.1/src/c/cz/mg/vulkan/PFN_vkGetPhysicalDeviceFormatProperties2KHR.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetPhysicalDeviceFormatProperties2KHR_callNative(JNIEnv* env, jclass jc, jlong address, jlong physicalDevice, jint format, jlong pFormatProperties){
    (void)env;
    (void)jc;
    PFN_vkGetPhysicalDeviceFormatProperties2KHR f = (PFN_vkGetPhysicalDeviceFormatProperties2KHR)jniLongToFunctionPointer(address);
    f(
        *((VkPhysicalDevice*)jniLongToPointer(physicalDevice)),
        (VkFormat)format,
        ((VkFormatProperties2*)jniLongToPointer(pFormatProperties))
    );
}

File: 1.1/src/cz/mg/vulkan/PFNvkGetPhysicalDeviceFormatProperties2KHR.java
package cz.mg.vulkan;

public class PFNvkGetPhysicalDeviceFormatProperties2KHR extends VkFunctionPointer {
    public PFNvkGetPhysicalDeviceFormatProperties2KHR() {
    }

    protected PFNvkGetPhysicalDeviceFormatProperties2KHR(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetPhysicalDeviceFormatProperties2KHR(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetPhysicalDeviceFormatProperties2KHR(long value) {
        setValue(value);
    }

    public PFNvkGetPhysicalDeviceFormatProperties2KHR(VkInstance instance) {
        super(instance, new VkString("vkGetPhysicalDeviceFormatProperties2KHR"));
    }

    public void call(VkPhysicalDevice physicalDevice, int format, VkFormatProperties2 pFormatProperties){
        callNative(getValue(), physicalDevice != null ? physicalDevice.getVkAddress() : VkPointer.getNullAddressNative(), format, pFormatProperties != null ? pFormatProperties.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long physicalDevice, int format, long pFormatProperties);

}
